lib/url.c

```diff
@@ -3283,6 +3284,12 @@ ConnectionExists(struct Curl_easy *data,
       pipeLen = check->send_pipe->size + check->recv_pipe->size;
 
       if(canPipeline) {
+        /*
+         * We can not use this connection if the connection is marked for
```

Comment:
comment is too many lines, though i know you were doing it like it is above. instead do it like
~~~c
        /* If the connection is marked for closure it can't be used. */
        if(check->bits.close)
          continue;
~~~
although it seems superfluous to me, because the code implies that. you could probably eschew it